### PR TITLE
fix: duplicate heart emojis [AR-2661]

### PR DESCRIPTION
**Pull request by Garzas**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/AR-2661" title="AR-2661" target="_blank">StoryAR-2661</a>  Reactions - First version for January
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
    - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
    - [x] contains a reference JIRA issue number like `SQPIT-764`
    - [x] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
    - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Because current web and iOS don't handle other emojis than the heart so we need temporarily disable other emojis. 
Also, we need to merge different hearts emojis into one, because other platforms are using an ugly ones 😅 
 "❤️" ❤"

### Dependencies (Optional)

_If there are some other pull requests related to this one (e.g. new releases of frameworks), specify them here._

Needs releases with:

- [ ] GitHub link to other pull request

#### Test Coverage (Optional)

- [ ] I have added automated test to this contribution

#### How to Test

_Briefly describe how this change was tested and if applicable the exact steps taken to verify that it works as expected._

### Notes (Optional)

_Specify here any other facts that you think are important for this issue._

### Attachments (Optional)

_Attachments like images, videos, etc. (drag and drop in the text box)_ 

| Before | After |
| ----------- | ------------ |
|  Screenshot 2022-11-04 at 10 39 41 |  Screenshot 2022-11-04 at 10 38 33 |

----
#### PR Post Submission Checklist for internal contributors (Optional)

- [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

- [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

**Comment by AndroidBob** [comment]
Build [1](https://10.10.124.134/job/wire-android-reloaded/job/PR-1140/1/) **failed**.

File: app/src/main/kotlin/com/wire/android/mapper/MessageMapper.kt [hunk 1]
```diff
@@ -50,8 +50,32 @@ class MessageMapper @Inject constructor(
...
         )
 
-        val footer = if(message is Message.Regular) {
-            MessageFooter(message.id, message.reactions.totalReactions, message.reactions.selfUserReactions)
+        val footer = if (message is Message.Regular) {
+            // TODO find ugly and proper heart emoji and merge them to ugly one 😅
+            val totalHeartsCount = message.reactions.totalReactions
+                .filterKeys { isHeart(it) }.values
+                .scan(0) { acc, count -> acc + count }
+                .lastOrNull()
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/mapper/MessageMapper.kt]: Kotlin's stdlib has a [sumOf](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.collections/sum-of.html) and for collections of numbers, [sum](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.sequences/sum.html) also exists to make things simpler.

```suggestion
                .sum()
```

File: app/src/main/kotlin/com/wire/android/mapper/MessageMapper.kt [hunk 2]
```diff
@@ -50,8 +50,32 @@ class MessageMapper @Inject constructor(
...
+            val hasSelfHeart = message.reactions.selfUserReactions.any { isHeart(it) }
+
+            MessageFooter(message.id,
+                message.reactions.totalReactions
+                    .filter { !isHeart(it.key) }
+                    .run {
+                        if (totalHeartsCount != null && totalHeartsCount != 0)
```
Comment (vitorhugods) [inline app/src/main/kotlin/com/wire/android/mapper/MessageMapper.kt]: With the suggestion above, totalHeartCount is no longer nullable.
```suggestion
                        if (totalHeartsCount != 0)
```

**Review by vitorhugods — CHANGES_REQUESTED** [review]
Just tiny suggestions to reduce the code a bit.

Nicely done.

**Comment by AndroidBob** [comment]
Build [2](https://10.10.124.134/job/wire-android-reloaded/job/PR-1140/2/) **failed**.

**Review by vitorhugods — APPROVED** [review]
👨‍🍳🤌

**Comment by AndroidBob** [comment]
Build [3](https://10.10.124.134/job/wire-android-reloaded/job/PR-1140/3/) **failed**.

**Comment by AndroidBob** [comment]
Build [4](https://10.10.124.134/job/wire-android-reloaded/job/PR-1140/4/) **failed**.

**Comment by AndroidBob** [comment]
Build [5](https://10.10.124.134/job/wire-android-reloaded/job/PR-1140/5/) **succeeded**.

The build produced the following APK's:
- [com.wire.android-vv0.0.1(4920126)(4920126)-dev-release.apk](https://z-lohika.s3-eu-west-1.amazonaws.com/megazord/android/reloaded/dev/release/com.wire.android-vv0.0.1(4920126)(4920126)-dev-release.apk)

**Pull request merged**